core/utils.py

```diff
@@ -1,23 +1,78 @@
 """Utility functions to load data from .mat .yaml and .h5 files."""
 
 import numpy as np
+from jax import numpy as jnp
 import h5py
 import os
 import yaml
 import scipy.io as spio
 import pickle
 from typing import Text
+from pynwb import NWBFile, NWBHDF5IO
+from ndx_pose import PoseEstimationSeries, PoseEstimation
 
+# Constants 
+MM_TO_M = 0.001
 
-def loadmat(filename):
+def daance_to_stac_mjx(data):
     """
-    this function should be called instead of direct spio.loadmat
-    as it cures the problem of not properly recovering python dictionaries
-    from mat files. It calls the function check keys to cure all entries
-    which are still mat-objects
+    Rearrange the data daance .mat mocap ordering
+    to internal order, and scale from millimeters to
+    meters. 
+
+    Args: mocap data arranged in dannce ordering, and
+    stored in mm.
+
+    Out: data rearranged into internal format, scaled to 
+    meters.
+    """
+
+    kp_names = params["KP_NAMES"]
+    # argsort returns the indices that would sort the array
+    stac_keypoint_order = np.argsort(kp_names)
```

Comment:
I'm not sure we want to be sorting these -- shouldn't the user just specify the ordering in the config?

Reply:
Makes sense to me, though I will consult with Charles to see if there is a reason he did it this way.

Reply:
According to Charles, the issue is that the keypoint order and the marker site mapping dict should be in the same order, since they are currently be subtracted from one another, element by element. If we don't do this, then the user must keep them in sync. Perhaps there is a better way to do this. (Sort them internally, but keep track of the unsorted order.) What do you think?

Reply:
I don't think we should sort. There's nothing intrinsically meaningful about alphabetical ordering.

We should have:
1. The list of keypoint names, preferably parsed from metadata in the mocap file
2. A list of site names from the mujoco body model XML
3. A mapping of keypoint names to site names, which is used to extract keypoints and reorder them to match the sites they correspond to.

An extra sorting step only adds a layer of potential failure modes.

Reply:
I agree. It's not immediately clear how to do this, so I am creating an issue to track this for a later date. 
https://github.com/talmolab/stac-mjx/issues/26